lib/inferno/dsl/fhir_evaluator/data_summary.rb

```diff
@@ -0,0 +1,96 @@
+module FhirEvaluator
+  # At this point DataSummary isn't actually used by the evaluator.
+  # We will bring back when we need it.
+
+  class DataSummary
+    attr_accessor :root_resource_ids, # All Example (root resource) Ids
+                  :root_bundle_resource_ids, # All Example (root resource) Ids that are Bundle
+                  :domain_resource_ids,   # Domain resource Ids from root and child resources (exclude Bundle)
+                  :resource_profile_map,  # Resources and corresponding profiles
+                  :resource_patient_map,  # Resources and corresponding Patient Ids as subject
+                  :resource_subject_map   # Resources and corresponding subject
+
+    def initialize(_data)
+      @root_resource_ids = []
+      @root_bundle_resource_ids = []
+      @domain_resource_ids = []
+      @resource_profile_map = []
+      @resource_patient_map = []
+      @resource_subject_map = []
+    end
+
+    def validate(data)
+      resource_ids = data.map { |r| extract_resources_ids(r) }.flatten
+
+      if resource_ids.uniq == resource_ids
+
+        # puts 'No duplicate Ids found. Proceed to evaluate..'
+      else
+        resource_ids.detect { |r| resource_ids.count(r) > 1 }
+        # puts "Warning: Found duplicate resource Ids: #{dup}. Please validate Examples before running FHIR Evaluator."
+      end
+    end
+
+    def summarize(data)
+      # @root_resource_ids = data.map { |r| { type: r.resourceType, id: r.id } }
+      # @root_bundle_resource_ids = data.map { |r| { type: r.resourceType, id: r.id } if r.resourceType == 'Bundle' }
+
+      # id_hash = Hash.new { |hash, key| hash[key] = [] }
+      # data.map { |e| resources(e) }.flatten.each do |item|
+      #   id_hash[item[:type]] << item[:id]
+      # end
+      # @domain_resource_ids = id_hash.to_a
+
+      # @resource_profile_map = data.map { |e| resources_profiles(e) }.flatten.uniq
+      # @resource_patient_map = data.map { |e| resources_patients(e) }.flatten.uniq
+      # @resource_subject_map = data.map { |e| resources_subjects(e) }.flatten.uniq
+    end
+
+    def extract_resources_ids(resource)
```

Comment:
These `extract_noun1_noun2` method names don't make sense to me. Is it extracting the first noun, the second noun, or both? They should be renamed.

Reply:
Per discussion with Dylan, we wll remove this file. The comment is archived at https://gitlab.mitre.org/inferno/fhir_evaluator/-/issues/1. We will come back to this when we decide to reuse.